src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/UploadAppealFormHandlerTest.java

```diff
@@ -0,0 +1,182 @@
+        "someurl",
+        "somebinaryurl",
+        "somefilename.pdf");
+
+    private final DocumentWithDescription someAppealFormDocument = new DocumentWithDescription(
+        someDoc,
+        "somedescription");
+
+    private final DocumentWithMetadata someTribunalMeta = new DocumentWithMetadata(
+        someDoc,
+        "some description",
+        "21/07/2021",
+        DocumentTag.APPEAL_SUBMISSION,
+        "some supplier"
+    );
+
+    private List<IdValue<DocumentWithDescription>> allAppealFormDocuments;
+    private List<IdValue<DocumentWithMetadata>> allTribunalDocuments;
+
+    @Mock
+    private Callback<AsylumCase> callback;
+    @Mock
+    private CaseDetails<AsylumCase> caseDetails;
+    @Mock
+    private AsylumCase asylumCase;
+    @Mock
+    private DocumentReceiver documentReceiver;
+    @Mock
+    private DocumentsAppender documentsAppender;
+    @Mock
+    private DocumentWithMetadata appealForm1WithMetadata;
+    private UploadAppealFormHandler uploadAppealFormHandler;
+
+    @BeforeEach
+    public void setUp() {
+        uploadAppealFormHandler =
+                new UploadAppealFormHandler(documentReceiver,documentsAppender);
+
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(callback.getCaseDetails().getCaseData()).thenReturn(asylumCase);
+        when(asylumCase.read(IS_ADMIN, YesOrNo.class)).thenReturn(Optional.of(YesOrNo.YES));
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Event.class,names = {"START_APPEAL"})
+    void should_append_appeal_forms_to_tribunal_documents(Event event) {
+
+        when(callback.getEvent()).thenReturn(event);
+
+        allAppealFormDocuments = Arrays.asList(
+            new IdValue<>("1", someAppealFormDocument)
+        );
+
+        allTribunalDocuments = Arrays.asList(
+            new IdValue<>("1", someTribunalMeta)
+        );
+
+        List<DocumentWithMetadata> docsWithMetadata = Arrays.asList(appealForm1WithMetadata);
+
+        when(asylumCase.read(APPELLANT_NAME_FOR_DISPLAY)).thenReturn(Optional.of("somename"));
+        when(asylumCase.read(APPEAL_REFERENCE_NUMBER)).thenReturn(Optional.of("DRAFT"));
+
+        when(asylumCase.read(UPLOAD_THE_APPEAL_FORM_DOCS)).thenReturn(Optional.of(allAppealFormDocuments));
+        when(asylumCase.read(TRIBUNAL_DOCUMENTS)).thenReturn(Optional.of(allTribunalDocuments));
+
+        when(documentReceiver.tryReceive(someAppealFormDocument, DocumentTag.APPEAL_FORM))
+            .thenReturn(Optional.of(appealForm1WithMetadata));
+
+        when(documentsAppender.prepend(allTribunalDocuments, docsWithMetadata)).thenReturn(allTribunalDocuments);
+
+        PreSubmitCallbackResponse<AsylumCase> callbackResponse = uploadAppealFormHandler.handle(ABOUT_TO_SUBMIT,callback);
+
+        assertThat(callbackResponse).isNotNull();
+
+        verify(documentsAppender, times(1)).prepend(
+            allTribunalDocuments,
+            docsWithMetadata
+        );
```

Comment:
- It can also test if the document is being renamed.

Reply:
I couldnt spend time on this, so will resolve this for now.

Reply:
As discussed, swapped this event to submit appeal so will take it up under a different ticket